src/spl/spline.h

```diff
@@ -236,8 +257,113 @@ class Spline {
     return total_length;
   }
 
+  std::vector<double> ComputeBezierDegreeElevationCoeffients(int dimension) const {
+    std::vector<double> alpha;
+    for (int i = 0; i < GetDegree(dimension).get() + 2; ++i) {
+      alpha.push_back(static_cast<double>(i) / (GetDegree(dimension).get() + 1));
+    }
+    return alpha;
+  }
+
+  std::vector<int> ProduceBezierSegments(int dimension) {
+    size_t position = GetKnotVector(dimension)->GetNumberOfKnots() - GetDegree(dimension).get() - 2;
+    std::vector<int> diff(GetKnotVector(dimension)->GetNumberOfDifferentKnots() - 2, GetDegree(dimension).get());
+    for (size_t i = GetKnotVector(dimension)->GetNumberOfDifferentKnots() - 2; i > 0; --i, --position) {
+      while (GetKnotVector(dimension)->GetKnot(position) == GetKnotVector(dimension)->GetKnot(position - 1)) {
+        --position;
+        --diff[i - 1];
+      }
+      InsertKnot(GetKnotVector(dimension)->GetKnot(position), dimension, static_cast<size_t>(diff[i - 1]));
+    }
+    return diff;
+  }
+
+  void RemoveBezierKnots(std::vector<int> diff, int dimension) {
+    size_t position = GetKnotVector(dimension)->GetNumberOfKnots() - GetDegree(dimension).get() - 2;
+    for (size_t i = GetKnotVector(dimension)->GetNumberOfDifferentKnots() - 2; i > 0; --i, --position) {
+      while (GetKnotVector(dimension)->GetKnot(position) == GetKnotVector(dimension)->GetKnot(position - 1)) {
+        --position;
+      }
+      RemoveKnot(GetKnotVector(dimension)->GetKnot(position), dimension,
+                 util::NumericSettings<double>::kEpsilon(), static_cast<size_t>(diff[i - 1] - 1));
+    }
+  }
+
+  virtual int GetBezierPointLength() const = 0;
+
+  virtual std::vector<double> GetBezierSegment(int dimension, int segment) const {
+    util::MultiIndexHandler<DIM> point_handler(GetPointsPerDirection());
+    int width = GetDegree(dimension).get() + 1;
+    int segment_length = GetNumberOfControlPoints() / GetPointsPerDirection()[dimension];
+    int point_length = this->GetBezierPointLength();
+    std::vector<double> bezier_cps(static_cast<size_t>(point_length * width * segment_length));
+    for (int i = 0; i < point_handler.Get1DLength(); ++i, ++point_handler) {
+      auto index_in_dir = point_handler[dimension];
+      if (index_in_dir >= segment * width && index_in_dir < (segment + 1) * width) {
+        double weight = this->GetWeight(point_handler.GetIndices());
+        if (point_length == GetPointDim() + 1) {
```

Comment:
(Partly) different implementations for non-rational & rational splines?

Reply:
Then there should be a different implementation of the whole method  as there would be too many function parameters when only distinguishing partly.